### PR TITLE
Prayer: Fixed time remaining text when time was greater than an hour

### PR DESCRIPTION
**Pull request by landychan**
The current implementation of calculating the time left on prayer only shows minutes remaining. When the time left on prayer was over an hour, it would cut off the hour and only show the minutes left.

I ran into this when I wore high prayer bonus (50+) and used a slow draining prayer (Preserve) with a high Prayer level.

I also added tests for the prayer text formatting :)

### CODE REVIEW

File: runelite-client/src/test/java/net/runelite/client/plugins/prayer/PrayerPluginTest.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+		when(client.isPrayerActive(Prayer.PRESERVE)).thenReturn(true);
+		when(client.getBoostedSkillLevel(Skill.PRAYER)).thenReturn(99);
+		when(client.getItemContainer(InventoryID.EQUIPMENT)).thenReturn(itemContainer);
+		when(itemContainer.getItems()).thenReturn(new Item[]{});
+
+		prayerPlugin.onItemContainerChanged(new ItemContainerChanged(InventoryID.EQUIPMENT.getId(), itemContainer));
+		String actualString = prayerPlugin.getEstimatedTimeRemaining(true);
+
+		Assert.assertEquals(expectedString, actualString);
```
Comment (Nightfirecat) [inline runelite-client/src/test/java/net/runelite/client/plugins/prayer/PrayerPluginTest.java:156]: Import `assertEquals`

File: runelite-client/src/test/java/net/runelite/client/plugins/prayer/PrayerPluginTest.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+				.slot(EquipmentInventorySlot.WEAPON.getSlotIdx())
+				.prayer(50)
+				.build());
+
+		ItemContainer itemContainer = mock(ItemContainer.class);
+
+		when(client.isPrayerActive(Prayer.PRESERVE)).thenReturn(true);
+		when(client.getBoostedSkillLevel(Skill.PRAYER)).thenReturn(99);
+		when(client.getItemContainer(InventoryID.EQUIPMENT)).thenReturn(itemContainer);
+		when(itemContainer.getItems()).thenReturn(new Item[]{new Item(ItemID.TWISTED_BOW, 1)});
```
Comment (Nightfirecat) [inline runelite-client/src/test/java/net/runelite/client/plugins/prayer/PrayerPluginTest.java:176]: Don't use a real item ID unless you're using real stats.

Comment (landychan) [reply runelite-client/src/test/java/net/runelite/client/plugins/prayer/PrayerPluginTest.java]: Replaced the real item ID with `99999`

File: runelite-client/src/test/java/net/runelite/client/plugins/prayer/PrayerPluginTest.java (hunk 1 above)
Comment (landychan) [reply runelite-client/src/test/java/net/runelite/client/plugins/prayer/PrayerPluginTest.java]: Imported `assertEquals` in the latest commit

File: runelite-client/src/test/java/net/runelite/client/plugins/prayer/PrayerPluginTest.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.inject.Inject;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static org.mockito.ArgumentMatchers.anyBoolean;
```
[the remark this answers is not in the document]
Comment (landychan) [reply runelite-client/src/test/java/net/runelite/client/plugins/prayer/PrayerPluginTest.java]: Fixed in the latest commit

File: runelite-client/src/test/java/net/runelite/client/plugins/prayer/PrayerPluginTest.java [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+	public void before()
+	{
+		Guice.createInjector(BoundFieldModule.of(this)).injectMembers(this);
+	}
+
+	@Test
+	public void testGetEstimatedTimeRemainingOverOneHour()
+	{
+
+		String expectedString = "1:19:12";
```
[the remark this answers is not in the document]
Comment (landychan) [reply runelite-client/src/test/java/net/runelite/client/plugins/prayer/PrayerPluginTest.java]: Fixed in the latest commit

File: runelite-client/src/test/java/net/runelite/client/plugins/prayer/PrayerPluginTest.java [hunk 5]
```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2017, Adam <Adam@sigterm.info>
+ * Copyright (c) 2018, Raqes <j.raqes@gmail.com>
```
Comment (Nightfirecat) [inline runelite-client/src/test/java/net/runelite/client/plugins/prayer/PrayerPluginTest.java:3]: Neither of these people are you--this should be changed.

**Pull request merged**